toolchain/lower/file_context.cpp

```diff
@@ -238,17 +239,8 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id)
           ? return_type
           : llvm::Type::getVoidTy(llvm_context());
 
-  std::string mangled_name;
-  if (SemIR::IsEntryPoint(sem_ir(), function_id)) {
-    // TODO: Add an implicit `return 0` if `Run` doesn't return `i32`.
```

Comment:
I wouldn't expect the name mangler to add a return. Maybe this TODO should remain here? Perhaps this code should even keep the `IsEntryPoint` call so that doing so is smooth?

Reply:
Rephrased the comment and applied it to the return type computation in the previous statement/just above this line.

The fix isn't really in the name mangling, it's in choosing the return type and in the code generation for the function body. So it seems a bit more accurate having the comment apply to the return type computation - though it's still a fair way from where the other part of the fix has to happen in the code generation.